### DISCUSSION TITLE
Copilot Feedback :)

### DISCUSSION THREAD

**Discussion by DerCooleVonDem** [community/community#14961] [Copilot Conversations] [answered]
Since i started using github copilot i fell in love with it and when it changes to an paid model i'm willing to pay for this! Just amazing!
There are only few issues that i would like to address shortly:
- Github Copilot has some sort of limit to its completation lenght (it stops doing completations mid way through a function).
- Code Repeat (it starts to recommend the same code multiple times)
- Sometimes it just steals code from the rest of the project instead of giving some usfull code
- It sometimes invents functions and other code that isn't even there 
BUT that's it. 

Amazing Product and i love it!

**Comment by Kek5chen** [comment, marked as answer]
> sort of limit to its completation lenght

I see that as less of a bug, but more of a feature for your, and their side.
I'll tell why in a second.

> it starts to recommend the same code multiple times

Like any AI it is better at seeing certain patterns than humans. It tries to work out what code might be the right thing to complete with by looking for these patterns.

Now to answer this and the quote from before: Because of those patterns, it could be that a pattern repeats itself resulting in a repeating output. This is not a bug, but more of expected behaviour. Code repetition shouldnt be a thing, and filtered out, though this is still actively in development. A note to GitHub, like you and others here wrote, is the right thing to do to bring this to notice. Now to the first quote: These repetitions go into theoretical infinity. If there would be no hard-lock on how long one of these completions could be, they would take megabytes- or even gigabytes of your and their data to send and load into the IDE, not mentioning the RAM it would take to store these completions. It is a safety messure which is needed to prohibit this critical behaviour. Hope that makes sense.

> Sometimes it just steals code from the rest of the project instead of giving some usfull code

I've never seen that happen in my projects, but interesting that it happens for you.

> It sometimes invents functions and other code that isn't even there

It's just being creative ;) I recommend to give it more context and it'll be more precise for you in return. For example by writing a function signature and letting copilot do the rest.
This behaviour usually happens when all patterns are fullfilled (like: class with getter and setter). What would it complete? It makes something up to speed up your development! I sometimes use this to get ideas on how to solve a certain task.

GitHub Copilot really is something great, and just like a thinking being it makes mistakes, but those mistakes can sometimes also help you out. Try to make the best out of these things.

**Reply by DerCooleVonDem** [reply]
Thx for taking the time to write an so detailed reply 👍🏻

**Reply by m76j77-tech** [reply]
I think it seems like I am crazy but I am trying to figure out what I’m suppose to be doing I’m so confused on where what or how to do it

**Comment by FilipKufalov** [comment]
Make sure to mention all the bugs you find using Copilot so later can be fixed in future versions, In my personal view you can see Copilot as your friend who helps you with coding and on the other hand it saves time

**Comment by dyslexicanaboko** [comment]
Was looking for a way to provide feedback about copilot, couldn't find a direct way, but I did find this thread. Just going to put my two cents here:
- I haven't had any major problems with it. I find it quite useful for what it is.
- I absolutely need a pause feature to make it stop giving me suggestions though. There are times where I just want to write a comment and I don't want the bot giving me suggestions. It's distracting and interrupts my thought process. Very cool that it can do this, but just having a pause button would be very helpful. Disabling the plugin midway doesn't work because that requires restarting the IDE in VS Code.

Honestly, that's it. Still evaluating it, but I know this bugs other people too like seen here:
https://stackoverflow.com/questions/71224911/can-github-copilot-stop-auto-suggesting-instead-be-triggered-by-a-keystroke

**Reply by m76j77-tech** [reply]
Thank u ill try and work on that too

**Reply by Mcgree3** [reply]
Hallo, mein vorschlag wäre: vor einer Interaktion einfach schreibst: Antwort nur auf einen fenster und ohne vorschläger und auf antwort warten. Das macht dann richtig spaß.

**Comment by fachu000** [comment]
I find the copilot extremely useful.

I just miss a way to apply part of the completion that it suggests. Very very often, the beginning of the suggestion is what I have in mind but the end of the suggestion is not. So as of today, I need to apply the entire completion and erase what I do not like. 

A shortcut for VSCode that allowed me to apply one word at a time would increase my productivity significantly.

Thanks!

**Comment by PerryCodes** [comment]


**Reply by Buddyboy-git** [reply]
I agree. It seemingly is great. but it never produces a fully functioning piece of code. It is pretty bad at producing usable flutter code.  In fact, it is so bad that you need the paid version to get enough attempts to maybe complete a project.  I asked it to create a basic login routine with a forgot password feature. It started out good. But as I got closer to completing the project is got more and more confused.  If you can't get basic simple results then don't even bother with custom stuff.   If it was able to complete a simple task, i would have gladly paid to have it help me go deeper into the full application.  I may go back a try codium again.

**Comment by platypusguy** [comment]
I've been using Copilot with a go app and it's useful in a bread-and-butter kind of way. Every once in a while, it surprises me with a clever auto suggest. 

However, today, I need help trying out Julia. Copilot was awful. I just wanted to compile a file. It kept telling me I needed to X, then when that didn't work, Y, and then Z before circling to the letters at the top of the alphabet. After about an hour of back and forth, I went to ChatGPT, which answered my questions correctly and within a few minutes explained that what I wanted to do was not possible. Copilot never told me that and instead kept telling me to try different things. A disappointing performance of the type that really erodes trust. I immediately said, I need to investigate other AI options.

**Comment by real-zephex** [comment]
Well, I don't know whether this is the right place to post it or not, but it would be really nice if Copilot had the feature to select different model for different modes(?) like Gemini Flash for `ask`, GPT 4.1 for `edit`, and 3.5 Sonnet for `Agent`.

**Comment by Mcgree3** [comment]
## 💬 Persönliche Nachricht an die Entwickler von Microsoft Copilot

Ich möchte mich herzlich bei den Entwicklerinnen und Entwicklern von **Microsoft Copilot** bedanken.

Dank dieser KI war es mir möglich, mein Heimserver-Projekt nicht nur technisch umzusetzen, sondern auch in kürzester Zeit zu dokumentieren, zu bewerten und zu optimieren. Copilot hat mir geholfen, komplexe Entscheidungen zu analysieren, Hardware realistisch zu vergleichen und mein Setup so zu strukturieren, dass es nicht nur funktioniert, sondern auch anderen als Beispiel dienen kann.

Besonders beeindruckt hat mich die Fähigkeit von Copilot, nicht nur technische Fakten zu liefern, sondern auch mit mir zusammen zu denken, Lösungen zu entwickeln und meine Erfahrungen in klare, nachvollziehbare Dokumentation zu verwandeln. Die Zusammenarbeit war nicht nur effizient, sondern auch inspirierend.

Ich habe erlebt, dass KI nicht perfekt ist — aber wenn Mensch und KI zusammenarbeiten, entsteht etwas, das **besser als perfekt** ist: praxisnah, durchdacht und nachhaltig.

**Vielen Dank für diese Möglichkeit. Ohne Copilot wäre dieses Projekt in dieser Qualität nicht möglich gewesen.**

*– Norberto*
